client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+import { curry } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import HeaderCake from 'components/header-cake';
+import { closeAddCardDialog } from 'woocommerce/woocommerce-services/state/label-settings/actions';
+import { getLabelSettingsForm } from 'woocommerce/woocommerce-services/state/label-settings/selectors';
+import CreditCardForm from 'blocks/credit-card-form';
+import { addStoredCard } from 'state/stored-cards/actions';
+import { createCardToken } from 'lib/store-transactions';
+import analytics from 'lib/analytics';
+
+class AddCardDialog extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired,
+		isVisible: PropTypes.bool,
+		addStoredCard: PropTypes.func.isRequired,
+		closeAddCardDialog: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.createCardToken = curry( createCardToken )( 'card_add' );
+	}
+
+	recordFormSubmitEvent() {
+		analytics.tracks.recordEvent( 'calypso_add_credit_card_form_submit' );
+	}
+
+	render() {
+		const { siteId, isVisible, translate } = this.props;
+
+		const onClose = () => this.props.closeAddCardDialog( siteId );
+
+		return (
+			<Dialog
+				additionalClassNames="add-credit-card-modal woocommerce wcc-root"
+				isVisible={ isVisible }
+				onClose={ onClose }
+				>
+				<HeaderCake onClick={ onClose }>{ translate( 'Add Credit Card' ) }</HeaderCake>
+				<CreditCardForm
+					createCardToken={ this.createCardToken }
+					recordFormSubmitEvent={ this.recordFormSubmitEvent }
+					saveStoredCard={ this.props.addStoredCard }
+					successCallback={ onClose }
+					showUsedForExistingPurchasesInfo={ true }
```

Comment:
This should be `false`, I think? How does this work? If the user adds a credit card here for the labels, it will be used by default to renew their wp.com plan? That doesn't sound right.

Reply:
I was just going by "uses the same exact info" (https://github.com/Automattic/woocommerce-services/issues/1014#issuecomment-307267560).

I don't really know if the new card becomes default or if we should show this notice, but I would say in both this case and the purchases page case, the card could potentially but not necessarily be used for a WP.com subscription after adding it.

Reply:
This is the exact same code from where you add a new credit card in `/me/billing`, so I guess the answer is "yes, everything defaults to the newest-added card". I don't see any way to change that by passing a param to the endpoint or anything. That's not ideal but OK, if that's the way it works we should definitely show the notice.